I know now why you cry: Shotgun additions

tenmillimaster opened the pull request:
```SUMMARY: Content "Adds several new shotguns, corrects stats on others"```

#### Purpose of change
I stared at the Leadworks defender too long and noticed some things.
The variety of shotguns in game felt lacking. There were few 'tacticool' variants, and historical and hunting ones felt missing too.

Any tube fed gun is probably too complicated for a survivor to make- the tube feed necessitates a shell lifter assembly and good fitment of parts, which is probably going to need a machine shop.

There's simply too much work to make an Abzats style gun work.

Arredondo speedloaders need to be indexed to actually be used speed-ily.
#### Describe the solution
Three Neosteads (L12 defenders) were imported into the United States, then later willingly relinquished to the ATF. As such the Defender has been obsoleted. It'll be migrated to the Tavor T12,  another gun which boasts a huge magazine capacity relative to its size.

The Lever action survivor shotgun has been axed. It'll be migrated to an 1887 Chiappa T-series, which is obviously a reference to Terminator 2.

Hunting variants with extra long barrels have been added for the 870 and 500. A breaching variant based on the 870 MCS system has been added as well. The militarized 590 has been added. The shotgun hunter starts with an 870 Wingmaster now.

A few Destructive Device/ tacticoolish shotguns from the early 90's/80's have been added- the USAS 12, the Streetsweeper and the SPAS-12. The latter two are fairly popular as DD shotguns go, especially the SPAS.  The USAS has been sporadically made stateside, but is more popular internationally. It's one of the very few full-auto shotguns, so it's been added as a rare swat shotgun.

2 Popular semi-auto shotguns, the Mossberg 930 and Browning Auto 5 have been added. 

The m1897 in a trench gun config has been added, with its ridiculously long bayonet as well.


There's now a magazine mod needed to use the speedloaders- it has a recipe as well.

The Abzats has been obsoleted, and the 'manual auto shotgun' has been changed into a beltfed gatling gun.
#### Describe alternatives you've considered
Migrating the Defender to something closer in size, but nothing really came to mind.
Giving the survivor lever action a magazine- like the saiga mag, but these aren't very common at all anymore.
I'd like to get around to some other shotguns sometime, the Remington 1100, the FN SLP, the Beretta 1301, the Ithaca 37, mossberg's horrid shotgun abortions (shockwave), keltec ks7... but without any meaningful differences between semi-auto and manual action, there's not much point currently.

I would have liked to add tube magazine extensions, but the ancient C++ magazine mods from 0.A days seem to have been deleted. 

I would have also liked to have the heat shield on the 590, the trench gun, and the SPAS 12, but I couldn't quite work out how to get the the get_hit() function to add bonus to hit from a mod, so I've held off on that as well.

I would like for there to be a new bayonet slot on guns- proper bayonets can mount there, that crappy spike bayonets that the player makes should mount on the underbarrel slot. This made sense to me on the 590 and the 1897.
#### Additional context

The ATF really actually has a letter out there stating how quickly the streetsweeper can be discharged.

[Relevant.](https://youtu.be/heJBfReQF9Q?t=79)


Some discussion with mlangsdorf on Discord, and a quick napkin calculation based on [ian mccollum's miss of a 28" tall pepper popper at 7 yards](https://youtu.be/tYjThckYqBQ?t=1063) yields a 'no-sights accuracy expectation' of ~ 300 MOA. Visually, with a man sized target at 100 yards:
![image](https://user-images.githubusercontent.com/7764202/66629105-6dd15c80-ebc6-11e9-90bd-6a9d9af8ce22.png)

Preliminary testing with the no-sights shotguns here showed that CDDA refused to accept 30000 as a dispersion value.  Even its max value, ~3000 iirc, felt 'off' in game when I was shooting at things 7 tiles away and missing by a mile.  1000 felt much more right.



EDIT:

Many thanks to Esotericist, Mlangsdorf, and LaVeyanFiend for their help in this.

Rail-Runner commented on data/json/items/gun/shot.json:
```diff
@@ -57,30 +35,43 @@
-    "//": "Hard to value this thing, but considering what a $2000 pricepoint puts it on par with, that is too much.",
-    "description": "A six-barrel hand-cranked automatic shotgun made from bicycle parts.  Though a bit unwieldy, it is exceedingly powerful for such a simple machine.",
+    "name": "12-gauge gatling gun",
+    "description": "An electrically driven six barrel gatling shotgun, fed from handmade cloth belts.  Even properly mounted, this seems like an unwieldy beast, and the six separate barrels make for difficult zeroing.  The externally driven action means this is much less likely to jam.",
```
Wouldn't this need to use UPS charges as well, if it's electrically driven?

Maybe also wouldn't necessarily be mounted then, if it's not hand-cranked; one could likely design such a device so as to be hip-fired.

tenmillimaster replied:
For all the reasons we didn't go very far with hip-fired M134's, I'm not doing this.

Rail-Runner replied:
Well, still needs to consume UPS charges either way then. So you have to have a charged battery on your vehicle if you want to fire it, just like M134 works currently.

tenmillimaster replied:
Sounds good!

Rail-Runner commented on data/json/items/magazine/shot.json:
```diff
@@ -32,6 +32,39 @@
+    "type": "MAGAZINE",
+    "name": "USAS-12 drum magazine",
+    "description": "A removable plastic magazine for the USAS-12 shotgun.  Holds 20 rounds.",
+    "weight": "1 g",
```
```suggestion
    "weight": "1 kg",
```
Probably a typo? Can't have 1 gram magazine.

tenmillimaster replied:
Ah thanks. it was a placeholder. I'll get that fixed.

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -240,20 +254,53 @@
+    "volume": "1427 ml",
+    "looks_like": "remington_870",
+    "ranged_damage": -5,
+    "skill": "pistol",
```
Suppressed shotgun bug
![image](https://user-images.githubusercontent.com/51099123/66661647-5a011700-ec15-11e9-8a0e-e4e723703a4c.png)
Seems it's either suppressed shotgun for muzzle mod or shrinking the gun if you still have the stock mount by swapping to shotgun skill.

tenmillimaster replied:
This is HILARIOUS. ok, the other shotgun pistol, the 12ga pistol has no muzzle location- I guess i'll delete its muzzle here.

tenmillimaster replied:
![image](https://user-images.githubusercontent.com/7764202/66671359-74d98880-ec21-11e9-980c-ff13c02be016.png)

Destragon replied:
Should the breacher really not be able to use a shotgun silencer? The tactical version can apparently take one in real life.
And I dunno if there are guidelines for this somewhere, but why do this (and the winchester 1887) use pistol skill? Just because they are a little shorter?

tenmillimaster replied:
They don't have stocks, and would presumably either be used at the waist or at an arms length. Putting it near your face is a bad idea.

Deleting the muzzle location is the easiest solution I could do to prevent caliber suppressors from being mounted.

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -331,53 +391,183 @@
+    "valid_mod_locations": [
+      [ "accessories", 4 ],
+      [ "barrel", 1 ],
+      [ "bore", 1 ],
```
Caliber conversion kits being un-obsoleted in the future confirmed? 🎉

tenmillimaster replied:
Mostly just an artifact of judicious copy and pasting.

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -331,53 +391,183 @@
+    "durability": 6,
+    "clip_size": 7,
+    "reload": 120,
+    "skill": "pistol",
```
Stealthy Terminator time?
![image](https://user-images.githubusercontent.com/51099123/66662558-0bed1300-ec17-11e9-884c-2326c19c47ba.png)

tenmillimaster replied:
![image](https://user-images.githubusercontent.com/7764202/66672032-f67de600-ec22-11e9-8d41-d2921f9d97c5.png)
Fixed, thanks!

tenmillimaster replied:
![image](https://user-images.githubusercontent.com/7764202/66674270-224f9a80-ec28-11e9-9a11-ad93ae7b31c2.png)

LaVeyanFiend replied:
Those are the same picture 🤔

tenmillimaster replied:
oh oops

tenmillimaster commented:
The speedloaders aren't working as expected: investigating.

tenmillimaster commented:
Many thanks to Kevin for helping out! The speedloader chute being a gunmod lets us bypass one of the 'valid speedloader?' on load checks. 

Trying to reload something small with somethign silly (8 round speedloader, 4 round tube), yields proper results. I'd prefer it if the extra rounds got dumped on the floor, but close enough.
![image](https://user-images.githubusercontent.com/7764202/66699022-d0028e00-eca8-11e9-84b0-323389499450.png)


 I've added the 1100, and in doing so noticed some of the volumes were off on the 500's and 870's. I've rechecked and updated their values.

There is now a new slot for loading port mods, where the chute will go. Mostly to prevent non-shotguns from using the magazine tube mod (and to save the magazine slot for  future mag-extension mods).



There was no documentation on what barrel_length was actually expecting, and with this funny little accident:
![image](https://user-images.githubusercontent.com/7764202/66698975-887c0200-eca8-11e9-8924-2bcaf99b981d.png) I've gone and poked around to find out. It's volume!

I've done some quick maths: 
![image](https://user-images.githubusercontent.com/7764202/66698962-5e2a4480-eca8-11e9-8ec0-adea8e2ad102.png)

9mL seems a decent number to lose when sawing off a barrel, per inch.

With the corrected numbers, the properties look close enough.
![image](https://user-images.githubusercontent.com/7764202/66699001-a9dcee00-eca8-11e9-90a7-8db3cce52b4b.png)

cainiaowu commented:
Hmm, I always thought LLC guns are fictional fancy guns that dont exist in real life, so maybe we could keep it instead of removing it from game?
You might want to ask the original creator, @Rivet-the-zombie , what do you think of it?

tenmillimaster commented:
From its comments, it's very clearly a neostead.

tenmillimaster commented:
I'm not sure if the failing tests are merge blockers, could anyone assist in interpretation please?

LaVeyanFiend commented on data/json/items/migration.json:
```diff
@@ -715,6 +715,20 @@
+    "id": "abzats",
+    "type": "MIGRATION",
+    "replace": "bigun"
+  },
```
```suggestion
  },
  {
```

tenmillimaster commented:
Failing tests appear to have been caused by abzats testing, which the test couldn't make. Replaced it with an m1014 in last commit.

LaVeyanFiend commented:
Looks like this is the test Travis is failing on:
![image](https://user-images.githubusercontent.com/51099123/67722056-8acfa300-f9ae-11e9-83ec-64e9830c4182.png)

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -90,13 +78,14 @@
-    "dispersion": 345,
+    "ranged_damage": 1,
+    "dispersion": 425,
+    "sight_dispersion": 1000,
```
Why a separate sight dispersion that's so high? No default dispersion is above 50 O_o

tenmillimaster replied:
The intent here is that these guns do not come with default sights of any kind.

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -238,22 +263,98 @@
+    "skill": "pistol",
+    "modes": [ [ "DEFAULT", "single", 1 ] ],
+    "//": "todo:update max supported range to 300MOA or find alt. soln.",
+    "sight_dispersion": 1000,
```
See previous comment

LaVeyanFiend commented on data/json/items/gun/shot.json:
```diff
@@ -331,60 +446,221 @@
-    "reload": 200,
-    "barrel_length": 4,
+    "clip_size": 16,
+    "sight_dispersion": 1000,
```
^

tenmillimaster commented:
After LVF's helpful playtesting, I need to work a little more on the base sight dispersion of an un-sighted gun; they're still missing a bit too much at close ranges.

tenmillimaster commented:
OK.

### New Item groups
I've pulled, to the best of my ability, relative prevalence for various aftermarket sights from gunbroker. These have been added, and are in  gunmods' item groups for future use.

These have been applied to the shotguns affected by lacking sights - which are conveniently all in `guns_shotguns_rare`- to keep the scope of this PR limited.

The use of `"EMPTY_GROUP"` in the new sight itemgroups appears to be the first in the repository, but meets the need I had here, to *occasionally* spawn these gunmods. 

Ideally, `contents-group` and `contents-string` would support optional probabilities, like below:
```json
      { "group": "guns_shotgun_rare", "prob": 22, "contents-group":{ [ "sights_shotgun_readied", "chance": 50 ] } },
```
50% would be the chance here that something from the group would spawn in this mockup, but currently only strings are parsed.

### Dispersion updates
Based on playtesting, I've backed off the sight dispersion of a referenceless, sightless firearm from 1000 to 750. I've also backed off the sight dispersion of a "M1913 Picatinny rail" equipped firearm from 1000 to 500.

This is close to what I feel should be the  new standards for firearms matching those descriptions. If there are no sights, no references, just a bare barrel, 750 feels appropriate.

If there are no sights, but a point of reference like a gutter sight, a vent rib, or an optics rail, 500 seems appropriate.

tenmillimaster commented on data/json/recipes/recipe_weapon.json:
```diff
@@ -3097,35 +3102,6 @@
   },
-  {
-    "type": "recipe",
-    "result": "abzats",
```
It's been added to recipes_obsolete. Is there anything else necessary?

ymber replied:
That's all it needs. I missed that going through it.

tenmillimaster commented on data/json/items/gun/shot.json:
```diff
@@ -331,60 +446,221 @@
+    "copy-from": "shotgun_base",
+    "type": "GUN",
+    "name": "Tavor TS12",
+    "description": "This is a triple tube magazine fed, gas-operated bullpup shotgun of Israeli Weapon Industries make.  It is capable of loading 15 shells, all in a relatively small package.  Like many other modern IWI designs, this looks more like a sci-fi prop gun than a firearm.  An integral top rail is provided for mounting sights.",
```
done

tenmillimaster commented:
>I found one new bug this introduced but generally it's most of the way there. Items with the fouling fault that are below the threshold to show as faulty have faulty in their names anyway.

Could you explain more? I'll ping you on discord.

tenmillimaster commented:
@ymber I've made the rest of your changes requested, but am unable to identify the origin of the bug you're experiencing.

ymber commented on data/json/items/gun/shot.json:
```diff
@@ -344,61 +458,224 @@
-    "id": "surv_levershotgun",
+    "id": "streetsweeper",
     "copy-from": "shotgun_base",
     "looks_like": "remington_870",
```
Duplicate `looks_like` is causing load errors.

ymber commented on data/json/items/gun/shot.json:
```diff
@@ -1,29 +1,7 @@
+    "copy-from": "pistol_base",
     "looks_like": "glock_17",
     "type": "GUN",
     "reload_noise_volume": 10,
```
Load errors from `reload_noise_volume` not updating the value from `pistol_base`.

Rivet-the-Zombie commented:
I don't like the removal of the L12 Defender at all. The Leadworks guns are fun, and not every firearm in the game has to be something that currently exists in the real world. Also, how does its removal affect the Home Frontier Starter Kit?

ymber commented:
`l_def_12` is still referred to by `No_Rivtech_Guns`.

ThomasLinkin commented:
i see the Remington 870 Breacher, and i wonder if i could use it in a certain way : to breach a door, not totally destroy it. if i wield it, and being chased by a horde of zombie, i could use it to blew up a door's lock in a single shot, not destroy it so that i could close it,block the horde, and proceed on escaping on the window. normally, as what i remembered, i need a few shot to blow a door, and completely destroy it, so it would be nice if this gun could actually BREACH a door.

tenmillimaster commented:
@Rivet-the-Zombie 

Add it back in, if you're so inclined. I removed it for the aforementioned reasons, and the inflexibility of built in mods would make it a poor seller IMO.
